internal/core/src/common/Schema.cpp

```diff
@@ -150,4 +150,36 @@ Schema::AbsentFields(Schema& old_schema) const {
     return std::make_unique<std::vector<FieldMeta>>(result);
 }
 
+const FieldMeta&
+Schema::GetFirstArrayFieldInStruct(const std::string& struct_name) const {
+    // Check cache first
+    auto cache_it = struct_array_field_cache_.find(struct_name);
```

Comment:
I was thinking of caching it in constructor. using this method we need to worry about concurrency safty.